homeassistant/components/hydrawise/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for the Hydrawise integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pydrawise import legacy
+from requests.exceptions import ConnectTimeout, HTTPError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
+from homeassistant.data_entry_flow import AbortFlow, FlowResult, FlowResultType
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_API_KEY): str})
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hydrawise."""
+
+    VERSION = 1
+
+    async def _create_entry(self, api_key: str) -> FlowResult:
+        """Create the config entry."""
+        try:
+            api = await self.hass.async_add_executor_job(
+                legacy.LegacyHydrawise, api_key
+            )
+        except ConnectTimeout:
+            await self._import_issue("timeout")
+            return self.async_abort(reason="timeout_connect")
```

Comment:
These should not abort. You should set an error and display it on the form.

Reply:
Actually I think it should abort if we're in YAML import mode, but show the error on the form otherwise? I changed it to do that instead.